pkg/scheduler/queue/tenant_queues_test.go

```diff
@@ -254,17 +252,38 @@ func TestQueuesRespectMaxTenantQueueSizeWithSubQueues(t *testing.T) {
 			if tq, ok := qb.tree.(*TreeQueue); ok {
 				assert.Equal(t, maxTenantQueueSize, tq.getNode(queuePath).ItemCount())
 			} else if itq, ok := qb.tree.(*MultiQueuingAlgorithmTreeQueue); ok {
-				assert.Equal(t, maxTenantQueueSize, itq.GetNode(queuePath).ItemCount())
+				var itemCount int
+				// if prioritizeQueryComponents, we need to build paths for each queue dimension
+				// and sum all items
+				if qb.prioritizeQueryComponents {
+					for _, addlQueueDim := range additionalQueueDimensions {
+						var path QueuePath
+						path = append(append(path, addlQueueDim...), "tenant-1")
+						if addlQueueDim == nil {
+							path = qb.makeQueuePathForTests("tenant-1")
+						}
+						itemCount += itq.GetNode(path).ItemCount()
+					}
+					assert.Equal(t, maxTenantQueueSize, itemCount)
+
+				} else {
+					assert.Equal(t, maxTenantQueueSize, itq.GetNode(queuePath).ItemCount())
+				}
 			}
 
-			// assert equal distribution of queue items between tenant node and 3 subnodes
+			// assert equal distribution of queue items between 4 subnodes
 			for _, v := range additionalQueueDimensions {
 				var checkPath QueuePath
 				if v == nil {
-					checkPath = append(QueuePath{"tenant-1"}, unknownQueueDimension)
-				} else {
+					v = []string{unknownQueueDimension}
+				}
+				switch qb.prioritizeQueryComponents {
+				case true:
+					checkPath = append(append(checkPath, v...), "tenant-1")
+				default:
```

Comment:
nits:
1. should we just replace the `0: nil,` from `additionalQueueDimensions` variable above with the unknown so we don't need this case? we only use unknown now anyway
2. switch on bool, let's do if-else

Reply:
1. We will still _receive_ requests that have nil `SchedulerRequest.AdditionalQueueDimensions`, so I think I prefer to keep it as nil to cover that case.
2. Fair enough!